index.js

```diff
@@ -70,6 +70,17 @@ Router.prototype.on = function on (method, path, opts, handler, store) {
   // handler validation
   assert(typeof handler === 'function', 'Handler should be a function')
 
+  // path ends with optional parameter
+  const optionalParamRegex = /(\/:[^/]*?)\?(\/?)$/
```

Comment:
moving this regexp near to `FULL_PATH_REGEXP` would avoid to recreate it for every route/`on` call. It should optimize a little bit the GC execution + remove the duplicated declaration*

Reply:
It would also avoid future bugs.

Reply:
Good idea, declaring it globally now.